Source/pack.cpp

```diff
@@ -92,8 +92,11 @@ void PackItem(ItemPack &packedItem, const Item &item, bool isHellfire)
 			packedItem.iCreateInfo = SDL_SwapLE16(item._iCreateInfo);
 			packedItem.bId = (item._iMagical << 1) | (item._iIdentified ? 1 : 0);
 			packedItem.bDur = item._iDurability;
-			packedItem.bMDur = item._iMaxDur;
-			packedItem.bCh = item._iCharges;
+			if (item._iMaxDur > 255)
+				packedItem.bMDur = 254;
+			else
+				packedItem.bMDur = item._iMaxDur;
+			packedItem.bCh = min<int32_t>(item._iCharges, packedItem.bMDur);

```

Comment:
This clamps `bCh` based on `bMDur`. I'm guessing you meant to set `bMDur` before `bDur` and clamp `bDur` instead.